pennylane_qiskit/vqe/vqe_runner.py

```diff
@@ -0,0 +1,548 @@
+            ansatz quantum circuit to use, default='EfficientSU2'
+        ansatz_config (dict): Optional, configuration parameters for the ansatz circuit if from Qiskit library.
+        x0 (array_like): Optional, initial vector of parameters.
+        optimizer (str): Optional, string specifying classical optimizer, default='SPSA'.
+        optimizer_config (dict): Optional, configuration parameters for the optimizer.
+        shots (int): Optional, number of shots to take per circuit, default=1024.
+        use_measurement_mitigation (bool): Optional, use measurement mitigation, default=False.
+        delete(bool): delete the program after using it, default=False.
+
+    Returns:
+        OptimizeResult: The result in SciPy optimization format.
+    """
+    # Init the dictionnaries
+    if ansatz_config is None:
+        ansatz_config = {}
+
+    if optimizer_config is None:
+        optimizer_config = {"maxiter": 100}
+
+    if not isinstance(hamiltonian, qml.Hamiltonian):
+        raise qml.QuantumFunctionError("Hamiltonian required.")
+
+    token = kwargs.get("ibmqx_token", None) or os.getenv("IBMQX_TOKEN")
+    url = kwargs.get("ibmqx_url", None) or os.getenv("IBMQX_URL")
+
+    if token is not None:  # pragma: no cover
+        # token was provided by the user, so attempt to enable an
+        # IBM Q account manually
+        def login():
+            ibmq_kwargs = {"url": url} if url is not None else {}
+            IBMQ.enable_account(token, **ibmq_kwargs)
+
+        active_account = IBMQ.active_account()
+        if active_account is None:
+            login()
+        else:
+            # There is already an active account:
+            # If the token is the same, do nothing.
+            # If the token is different, authenticate with the new account.
+            if active_account["token"] != token:
+                IBMQ.disable_account()
+                login()
+    else:
+        # check if an IBM Q account is already active.
+        #
+        # * IBMQ v2 credentials stored in active_account().
+        #   If no accounts are active, it returns None.
+
+        if IBMQ.active_account() is None:
+            # no active account
+            try:
+                # attempt to load a v2 account stored on disk
+                IBMQ.load_account()
+            except IBMQAccountError:
+                # attempt to enable an account manually using
+                # a provided token
+                raise IBMQAccountError(
+                    "No active IBM Q account, and no IBM Q token provided."
+                ) from None
+
+    options = {"backend_name": backend}
+
+    inputs = {}
+
+    # Extract the number of qubit from the hamiltonian
+    _, observables = hamiltonian.terms
+
+    qubit_set = set()
+    for obs in observables:
+        for qubit in obs.wires.tolist():
+            qubit_set.add(qubit)
+
+    num_qubits_h = list(qubit_set)[-1] + 1
```

Comment:
This is the intended behavior, because the circuit might not be empty where the hamiltonian does not act. The number of qubit in the circuit is 10 in your example. But I might choose a better variable name..